SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -0,0 +1,555 @@
+              Gust[GustDir] = gust_amp * (sin(2 * PI_NUMBER * x_gust));
+
+              // Gust derivatives
+              // dgust_dx = gust_amp*2*PI_NUMBER*(cos(2*PI_NUMBER*x_gust))/L;
+              // dgust_dy = 0;
+              // dgust_dt = gust_amp*2*PI_NUMBER*(cos(2*PI_NUMBER*x_gust))*(-Uinf)/L;
+            }
+            break;
+
+          case ONE_M_COSINE:
+            // Check if we are in the region where the gust is active
+            if (x_gust > 0 && x_gust < n) {
+              Gust[GustDir] = gust_amp * (1 - cos(2 * PI_NUMBER * x_gust));
+
+              // Gust derivatives
+              // dgust_dx = gust_amp*2*PI_NUMBER*(sin(2*PI_NUMBER*x_gust))/L;
+              // dgust_dy = 0;
+              // dgust_dt = gust_amp*2*PI_NUMBER*(sin(2*PI_NUMBER*x_gust))*(-Uinf)/L;
+            }
+            break;
+
+          case EOG:
+            // Check if we are in the region where the gust is active
+            if (x_gust > 0 && x_gust < n) {
+              Gust[GustDir] = -0.37 * gust_amp * sin(3 * PI_NUMBER * x_gust) * (1 - cos(2 * PI_NUMBER * x_gust));
+            }
+            break;
+
+          case VORTEX:
+
+            /*--- Use vortex distribution ---*/
+            // Algebraic vortex equation.
+            for (unsigned long i = 0; i < nVortex; i++) {
+              su2double r2 = pow(x - (x0[i] + Uinf * (Physical_t - tbegin)), 2) + pow(y - y0[i], 2);
+              su2double r = sqrt(r2);
+              su2double v_theta = vort_strenth[i] / (2 * PI_NUMBER) * r / (r2 + pow(r_core[i], 2));
+              Gust[0] = Gust[0] + v_theta * (y - y0[i]) / r;
+              Gust[1] = Gust[1] - v_theta * (x - (x0[i] + Uinf * (Physical_t - tbegin))) / r;
+            }
+            break;
+
+          case NONE:
+          default:
+
+            /*--- There is no wind gust specified. ---*/
+            if (rank == MASTER_NODE) {
+              cout << "No wind gust specified." << endl;
+            }
+            break;
+        }
+      }
+
+      /*--- Set the Wind Gust, Wind Gust Derivatives and the Grid Velocities ---*/
+
+      GustDer[0] = dgust_dx;
+      GustDer[1] = dgust_dy;
+      GustDer[2] = dgust_dt;
+
+      solver[iMGlevel][FLOW_SOL]->GetNodes()->SetWindGust(iPoint, Gust);
+      solver[iMGlevel][FLOW_SOL]->GetNodes()->SetWindGustDer(iPoint, GustDer);
+
+      GridVel = geometry[iMGlevel]->nodes->GetGridVel(iPoint);
+
+      /*--- Store new grid velocity ---*/
+
+      for (iDim = 0; iDim < nDim; iDim++) {
+        NewGridVel[iDim] = GridVel[iDim] - Gust[iDim];
+        geometry[iMGlevel]->nodes->SetGridVel(iPoint, iDim, NewGridVel[iDim]);
+      }
+    }
+  }
+
+  delete[] Gust;
+  delete[] GustDer;
+  delete[] NewGridVel;
+}
+
+void CFluidIteration::InitializeVortexDistribution(unsigned long& nVortex, vector<su2double>& x0, vector<su2double>& y0,
+                                                   vector<su2double>& vort_strength, vector<su2double>& r_core) {
+  /*--- Read in Vortex Distribution ---*/
```

Comment:
Is this needed? I thought we had some vortex stuff in the MMS @economon ?

Reply:
If I recall, that routine is related to the old implementation of wind gust features, which pre-dates the verification classes. Those initial conditions could probably be moved there (verification), or perhaps @talbring had something in mind with the modules for that

Reply:
You are right, for a minute I thought it had something to do with the Taylor-Green vortex.